packages/smooth_app/lib/helpers/picture_capture_helper.dart

```diff
@@ -13,32 +19,52 @@ Future<bool> uploadCapturedPicture(
   required Uri imageUri,
 }) async {
   final AppLocalizations appLocalizations = AppLocalizations.of(context);
-  final SendImage image = SendImage(
-    lang: ProductQuery.getLanguage(),
+  final LocalDatabase localDatabase = context.read<LocalDatabase>();
+  final String uniqueId = _getUniqueId(imageField, barcode);
+  final BackgroundImageInputData backgroundImageInputData =
+      BackgroundImageInputData(
+    processName: IMAGE_UPLOAD_TASK,
+    uniqueId: uniqueId,
     barcode: barcode,
-    imageField: imageField,
-    imageUri: imageUri,
+    imageField: imageField.value,
+    imageUri: File(imageUri.path).path,
+    languageCode: ProductQuery.getLanguage().code,
+    user: jsonEncode(ProductQuery.getUser().toJson()),
+    country: ProductQuery.getCountry()!.iso2Code,
   );
-  final Status? result = await LoadingDialog.run<Status>(
-    context: context,
-    future: OpenFoodAPIClient.addProductImage(
-      ProductQuery.getUser(),
-      image,
+  await TaskManager().addTask(
+    Task(
+      data: backgroundImageInputData.toJson(),
+      uniqueId: uniqueId,
+    ),
+  );
+
+  localDatabase.notifyListeners();
+  // ignore: use_build_context_synchronously
+  ScaffoldMessenger.of(context).showSnackBar(
```

Comment:
You can add a check if (monted) for the context

Reply:
Don't know but I can't seemingly use the `mounted` variable in this page

Reply:
One way to fix, is to store the context in a local variable.
It will thus allow you to remove // ignore lines